### PR TITLE
feat: kubeflow component for model version update

### PR DESCRIPTION
**Pull request by michalmrazek**
## Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

**Review by jsilva — CHANGES_REQUESTED** [review]

File: src/wanna/components/kubeflow/upload_model_version.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        "google-cloud-aiplatform==1.15.1",
+    ],
+)
+def upload_model_version(
+    project: str,
+    location: str,
+    display_name: str,
+    serving_container_image_uri: str,
+    artifact_uri: str,
+    model: dsl.Output[dsl.Artifact],
```
[the remark this answers is not in the document]
Comment (michalmrazek) [reply src/wanna/components/kubeflow/upload_model_version.py:19]: got it working with dsl.Model :)

File: src/wanna/components/kubeflow/upload_model_version.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    labels: Dict[str, str] = {},
+    version_aliases: List[str] = None,
```
[the remark this answers is not in the document]
Comment (michalmrazek) [reply src/wanna/components/kubeflow/upload_model_version.py:21]: kubeflow unfortunately does not understand Optional

File: src/wanna/components/kubeflow/upload_model_version.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    metadata: List[str] = None,
+    model_description: str = None,
+    version_description: str = None,
+):
+
```
[the remark this answers is not in the document]
Comment (michalmrazek) [reply src/wanna/components/kubeflow/upload_model_version.py:26]: 👍

File: src/wanna/components/kubeflow/upload_model_version.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        parent_model=parent_model_resource_name,
+        parent=parent,
+        model=vertex_model,
+    )
+    operation = client.upload_model(request=request)
+    response = operation.result()
+    new_model_resource_name = response.model
+
+    model.uri = vertex_uri_prefix + new_model_resource_name
+    model.metadata = {"resourceName": new_model_resource_name}
```
[the remark this answers is not in the document]
Comment (michalmrazek) [reply src/wanna/components/kubeflow/upload_model_version.py:82]: 👍

**Pull request merged**